### PR TITLE
AI Assistant: add upgrade notice to AI Assistant

### LINKED ISSUE #30736: AI Assistant: Add upgrade banner to block whenever user runs out of requests
Blocked by [#30726](https://github.com/Automattic/jetpack/issues/30726), we need to build the requests counter first. 

Once the user runs out of requests, add a way to let the user upgrade or buy more credits. 

As a first iteration, maybe we can use something similar to what we have in VideoPress, but with a link to the interstitial for Jetpack AI:

Screenshot 2023-05-16 at 11 36 57

### PR DESCRIPTION
**Pull request by nunyvega**
Fixes #30736

## Proposed changes:

* Add upgrade notice to the AI Assistant block. It won't be used yet.
It will be enabled once we have the monetization logic. 

Screenshot 2023-05-18 at 13 23 37

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply to your local JP
* Enable it by removing the `false && `
* confirm that it is visible on your editor
* Confirm that it redirects to the Jetpack AI interstitial

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/*
+ * Internal dependencies
+ */
+import { Nudge } from '../../shared/components/upgrade-nudge';
+
+const UpgradePrompt = () => {
+	return (
+		<Nudge
+			buttonText={ 'Upgrade' }
+			checkoutUrl={ 'admin.php?page=my-jetpack#/add-jetpack-ai' }
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js]: You can use the `Jetpack_Editor_Initial_State` global object here:

```suggestion
			checkoutUrl={ `${ Jetpack_Editor_Initial_State?.adminUrl } ?page=my-jetpack#/add-jetpack-ai` }
```

Something like this ^

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+/*
+ * Internal dependencies
+ */
+import { Nudge } from '../../shared/components/upgrade-nudge';
+
+const UpgradePrompt = () => {
+	return (
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js]: We should return `null` when it isn't a Jetpack site (at least for now)

```suggestion
const UpgradePrompt = () => {
	if ( isAtomicSite() || isSimpleSite() ) {
		return null;
	}
	
	return (
```

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js]: You need to requite the functions in the `External dependencies` block, too.

```es6
import { isAtomicSite, isSimpleSite } from '@automattic/jetpack-shared-extension-utils';
```

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js (hunk 1 above)
Comment (oskosk) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js]: Addressed in 3481fba980f7cab15851c51cf56e401c587bc1d9

**Review by oskosk — APPROVED** [review]
LGTM!

**Comment by oskosk** [comment]
LGTM but we need to revisit the styles

**Pull request merged**